src/AutoRest.CSharp/MgmtTest/Generation/ResourceTestWriter.cs

```diff
@@ -0,0 +1,214 @@
+                        MgmtClientOperation CreateOperation = HasGetExample(Context, _resource.ResourceCollection!) ? _resource.ResourceCollection!.GetOperation! : _resource.ResourceCollection!.CreateOperation!;
+                        _collectionTestWriter!.WriteGetCollection(_resource.ResourceCollection!.CreateOperation!, exampleGroup.Examples.First(), async);
+                        _writer.Append($"var createOperation = ");
+                        _collectionTestWriter!.WriteInvokeExampleInstanceMethod(_resource.ResourceCollection!.CreateOperation!, async, "CreateOrUpdate", exampleGroup.Examples.First());
+                        var respType = GenResponseType(_resource.ResourceCollection!.CreateOperation!, true, "CreateOrUpdate");
+                        var resultTypeDataName = _resource.ResourceCollection!.CreateOperation!.IsLongRunningOperation() ? ((Mgmt.Output.MgmtLongRunningOperation)respType.Arguments[0].Implementation).ResultType?.Name : ((Mgmt.Output.NonLongRunningOperation)respType.Arguments[0].Implementation).ResultDataType?.Name;
+                        if (resultTypeDataName != $"{_resource.Type.Name}Data" && _resource.ResourceCollection!.GetOperation!.ReturnType?.Name == $"{_resource.Type.Name}Data") {
+                            _writer.Line($"return collection.Get(createOperation.Value.Name);");
+                        }
+                        else {
+                            _writer.Line($"return createOperation.Value;");
+                        }
+                        break;
+                    }
+                }
+                else
+                {
+                    throw new Exception("// TODO: create/get resource when there is no create example");
+                }
+            }
+        }
+
+        public void WriteGetResource(Resource resource, bool async)
+        {
+            var resourceVariableName = useVariableName(_resource.Type.Name.FirstCharToLowerCase());
+            _writer.Line($"var {resourceVariableName} = {GetAwait(async)} Get{_resource.Type.Name}{GetAsyncSuffix(async)}();");
+            if (resource != _resource) {
+                var childResourceVariableName = useVariableName(resource.Type.Name.FirstCharToLowerCase());
+                _writer.Line($"var {childResourceVariableName} = {resourceVariableName}.Get{resource.Type.Name}();");
+            }
+        }
+
+        protected override CSharpType? WrapResourceDataType(CSharpType? type, MgmtRestOperation operation)
+        {
+            if (!IsResourceDataType(type, operation))
+                return type;
+
+            return _resource.Type;
+        }
+
+        protected override bool IsResourceDataType(CSharpType? type, MgmtRestOperation operation)
+        {
+            return _resource.Type.Equals(type);
+        }
+
+        protected void WriteMethodTest(Resource resource, MgmtClientOperation clientOperation, bool async)
+        {
+            Debug.Assert(clientOperation != null);
```

Comment:
This is redundant since the `clientOperation` can never be `null` because its type is non-nullable